### PR TITLE
History: display Address Book names for known destination addresses

### PR DESCRIPTION
**Pull request by xiphon**
Closes https://github.com/monero-project/monero-gui/issues/2757

monero-gui-history-names

### CODE REVIEW

**Comment by selsta** [comment]
I wonder if Sent / Received could still be there + the address book name. Unless you know what the green / orange mark means it is not clear if the transaction is incoming / outgoing.

But changing it to `Sent to John Doe` would take up a lot of space.

File: pages/History.qml [hunk 1]
```diff
@@ -633,7 +633,17 @@ Rectangle {
                                 MoneroComponents.TextPlain {
                                     font.family: MoneroComponents.Style.fontRegular.name
                                     font.pixelSize: 15
-                                    text: isout ? qsTr("Sent") : qsTr("Received") + translationManager.emptyString
+                                    text: {
+                                        if (!isout) {
+                                            return qsTr("Received") + translationManager.emptyString;
+                                        }
+                                        const addressBookName = currentWallet.addressBook.getDescription(address);
```
Comment (selsta) [inline pages/History.qml]:
```
2020-02-15 15:11:22.911	W qrc:/pages/History.qml:640: TypeError: Cannot read property 'addressBook' of undefined
```
shows up when closing the wallet.

Comment (xiphon) [reply pages/History.qml]: Updated

**Comment by xiphon** [comment]
> I wonder if Sent / Received could still be there + the address book name. Unless you know what the green / orange mark means it is not clear if the transaction is incoming / outgoing.
> 
> But changing it to `Sent to John Doe` would take up a lot of space. Not sure what the best solution is here.

Feel free to propose any other design option.

**Comment by rating89us** [comment]
I can't test it right now, but have you considered that a single address can correspond to multiple contacts in address book (#2595)?
In this case I believe it should return the contacts separated by comma: "Alice, Bob"

**Comment by xiphon** [comment]
> we should stop allowing multiple contacts to have the same address

We can't.

> have you considered that a single address can correspond to multiple contacts in address book?

Yes. It is intended to show the most recently added contact.

**Pull request merged**